lib/rubocop/cop/style/guard_clause.rb

```diff
@@ -157,11 +157,13 @@ def register_offense(node, scope_exiting_keyword, conditional_keyword)
           add_offense(node.loc.keyword, message: format(MSG, example: example))
         end
 
-        def guard_clause_source(guard_clause)
+        def guard_clause_exiting_keyword(guard_clause)
           parent = guard_clause.parent
 
-          if parent.and_type? || parent.or_type?
+          if parent.and_type?
             guard_clause.parent.source
+          elsif parent.or_type?
+            "return #{guard_clause.parent.source}"
```

Comment:
Looks like we need `return` on the `and` case too.

I think this is not related to `and` or `or`.
It seems that the same problem exists in `and`.

The following two codes can have different results.

```ruby
if something
  work and return
else
  test
end
```

```ruby
work and return if something
test
```

It is the following case:

* `something` : `true`
* `work`: `false`

```ruby
if true
  false and return
else
  p "hoge"
end
=> (no output)
```

```ruby
false and return if true
p "hoge"
=> "hoge"
```

Reply:
@daipom Thank you for your comment !
I think your code is correct.
I'll check it.

Reply:
I fixed this commit.
d1f0a224fce9e6f351fbae5e709b95fb00e92489

Reply:
Hmm, indeed it is...

Given this example, perhaps we should recommend this trailing-if-style (such as `work if cond`) only when we are sure that a `raise` or `return` will occur in the block.

It would be correct not to recommend this if it is unclear whether a `raise` or `return` will definitely occur in the block.